### PR TITLE
#160495513 Ch-mocha-test1

### PR DESCRIPTION
**Pull request by Johnpaul-Borree**
#### What does this PR do?
Test Api endpoint for create order
#### Description of Task to be completed?
- Create file to test Endpoint: Post /ap1/v1/orders
- Create tests to test for creating order taking sample order
- create tests for validating this endpoints
#### How should this be manually tested?
- Run command npm run tests 
- check continuous integrations 
- look out for readme badge on this repo
#### What are the relevant pivotal tracker stories?
#160495513

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        "quotes": [
+            "error",
+            "single"
+        ],
+        "semi": [
+            "error",
+            "always"
+        ]
+    }
+}
```
Comment (KaiserPhemi) [inline .eslintrc.json]: #### Nitpick
Always add a new line after the last line

File: src/controller/ordersControllers.js [hunk 2]
```diff
@@ -3,35 +3,35 @@ import orders from '../models/orders';
 
 
 export function postOrder (req, res) {
```
Comment (KaiserPhemi) [inline src/controller/ordersControllers.js]: #### Critical
You were advised to use `ES6`. Please remove every instances of `function` keyword and use the arrow function. i.e. This should be refactored to:
```javascript
export const postOrder = (req, res) => {
  // ....
}
```

Exceptions might be configuration files.

File: src/controller/ordersControllers.js [hunk 3]
```diff
@@ -3,35 +3,35 @@ import orders from '../models/orders';
...
+		quantities: req.body.quantities,
+		unitPrice: Math.floor( Math.random() + 200),
+		total: 'N' + req.body.quantities * (Math.floor( Math.random() + 200))
+	};
 
-    orders.push(order);
+	orders.push(order);
 
-    res.json(order);
+	res.json({justAdded: order, message: 'order Created'});
```
Comment (KaiserPhemi) [inline src/controller/ordersControllers.js]: #### Nitpick 
For objects, you should add a space between the curly braces
i.e.
```js
({ justAdded: order, message: 'order Created' })
```
Note the spaces between the contents and the curly braces. 
Also, go through your code and fix every instances of this.

File: src/controller/ordersControllers.js [hunk 4]
```diff
@@ -3,35 +3,35 @@ import orders from '../models/orders';
 
 
 export function postOrder (req, res) {
-    const {error} = validateOrders(req.body);
+	const {error} = validateOrders(req.body);
     
-    if(error) return res.status(400).json({status: 'success', message: error.details[0].message});
+	if(error) return res.status(400).json({status: 'Failed to create order', message: error.details[0].message});
```
Comment (KaiserPhemi) [inline src/controller/ordersControllers.js]: #### Nitpick
Please refactor this code and make it readable. It's obvious you aren't following the airbnb style guide.
For context, please check [this](https://github.com/airbnb/javascript#control-statements)

File: src/controller/ordersControllers.js [hunk 5]
```diff
@@ -3,35 +3,35 @@ import orders from '../models/orders';
...
+	};
 
-    orders.push(order);
+	orders.push(order);
 
-    res.json(order);
+	res.json({justAdded: order, message: 'order Created'});
 }
 
 function validateOrders(order) {
```
Comment (KaiserPhemi) [inline src/controller/ordersControllers.js]: #### Critical
Same thing I said above

File: src/controller/ordersControllers.js [hunk 6]
```diff
@@ -3,35 +3,35 @@ import orders from '../models/orders';
...
+		type: Joi.string().min(3).required(),
+		orderDate: Joi.string().min(6).required(),
+		expires: Joi.string().min(6).required(),
+		quantities: Joi.number().integer().min(1).max(50000).required(),
+	};
 
-    return Joi.validate(order, schema);
+	return Joi.validate(order, schema);
     
 }
```
Comment (KaiserPhemi) [inline src/controller/ordersControllers.js]: #### Nitpick
Always add a new line after the last line of code

**Review by KaiserPhemi — CHANGES_REQUESTED** [review]
Please go through the code and implement feedback.

File: .eslintrc.json (hunk 1 above)
Comment (Johnpaul-Borree) [reply .eslintrc.json]: Ok I will do that

File: src/controller/ordersControllers.js (hunk 2 above)
Comment (Johnpaul-Borree) [reply src/controller/ordersControllers.js]: Okay I will remove all instances of function keyword,
 thank you for suggesting a way

Comment (Johnpaul-Borree) [reply src/controller/ordersControllers.js]: I have implemented the feedback on this line, thank you

File: src/controller/ordersControllers.js (hunk 3 above)
Comment (Johnpaul-Borree) [reply src/controller/ordersControllers.js]: Thanks so much, I will fix this

Comment (Johnpaul-Borree) [reply src/controller/ordersControllers.js]: I just added a linting rule for this and corrected all appearances on this branch

File: src/controller/ordersControllers.js (hunk 5 above)
Comment (Johnpaul-Borree) [reply src/controller/ordersControllers.js]: Thanks, I have resolved as above

File: src/controller/ordersControllers.js (hunk 6 above)
Comment (Johnpaul-Borree) [reply src/controller/ordersControllers.js]: Thanks, I have corrected this

File: src/controller/ordersControllers.js (hunk 4 above)
Comment (Johnpaul-Borree) [reply src/controller/ordersControllers.js]: I just implemented this, thanks

**Comment by Johnpaul-Borree** [comment]
can I merge branch? @KaiserPhemi

**Pull request merged**